### PR TITLE
Online DDL CI: consolidated revertible and revert CI tests (part 1)

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Similarly to https://github.com/vitessio/vitess/pull/12061 and https://github.com/vitessio/vitess/pull/12055, we now consolidate `onlineddl_revertible` into `onlineddl_revert`.
After this has been merged, we can remove the `Required` tag from `onlineddl_revertible` and then followup with a romval PR, much like https://github.com/vitessio/vitess/pull/12182

This will free two CI workflows.

## Related Issue(s)

#6926

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go [hunk 1]
```diff
@@ -260,6 +205,261 @@ func TestSchemaChange(t *testing.T) {
...
+
+	removeBackticks := func(s string) string {
+		return strings.Replace(s, "`", "", -1)
+	}
+
+	for _, testcase := range testCases {
+		t.Run(testcase.name, func(t *testing.T) {
+
+			// func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, expectHint string, expectError string, skipWait bool) (uuid string) {
+			// func testOnlineDDLStatement(t *testing.T, alterStatement string, ddlStrategy string, executeStrategy string, tableName string, expectHint string) (uuid string) {
```
Comment (mattlord) [inline go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go:352]: We probably want to remove these?

File: go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go [hunk 2]
```diff
@@ -260,6 +205,261 @@ func TestSchemaChange(t *testing.T) {
...
+		},
+	}
+
+	var (
+		createTableWrapper = `CREATE TABLE onlineddl_test(%s)`
+		dropTableStatement = `
+			DROP TABLE onlineddl_test
+		`
+		tableName   = "onlineddl_test"
+		ddlStrategy = "online -declarative -allow-zero-in-date"
```
Comment (mattlord) [inline go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go:341]: I know this is internal, but IMO better for all of us to get into the habit of using double dashes everywhere.

File: go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go [hunk 3]
```diff
@@ -260,6 +205,261 @@ func TestSchemaChange(t *testing.T) {
...
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+				checkTable(t, tableName, false)
+			})
+
+			t.Run("create from-table", func(t *testing.T) {
+				fromStatement := fmt.Sprintf(createTableWrapper, testcase.fromSchema)
+				uuid := testOnlineDDLStatement(t, fromStatement, ddlStrategy, "vtgate", tableName, "")
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusComplete)
+				checkTable(t, tableName, true)
+			})
```
Comment (mattlord) [inline go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go:365]: It's not obvious to me how these are confirming it's revertible, but assuming it is (you would certainly know).

**Review by mattlord — APPROVED** [review]
❤️

File: go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go (hunk 1 above)
Comment (shlomi-noach) [reply go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go:352]: Whoops, thank you, fixed.

File: go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go (hunk 2 above)
Comment (shlomi-noach) [reply go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go:341]: Agreed! Fixed.

File: go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go (hunk 3 above)
Comment (shlomi-noach) [reply go/test/endtoend/onlineddl/revert/onlineddl_revert_test.go:365]: Ah. This is just preparation. Per test case, we (re)create the table, then run a migration, then see if the migration is revertible. Added comments to clarify.

**Review by deepthi — APPROVED** [review]
💯

**Pull request merged**